internal/controllers/machine/machine_controller.go

```diff
@@ -507,6 +507,15 @@ func (r *Reconciler) isDeleteNodeAllowed(ctx context.Context, cluster *clusterv1
 		return errClusterIsBeingDeleted
 	}
 
+	// If the machine is detached, considered as no nodeRef.
+	if machine.Annotations != nil {
+		if _, found := machine.Annotations[clusterv1.MachineDetachedAnnotation]; found {
```

Comment:
I'm a little confused about the exact goal here - if I understand correctly this annotation will result in a delete action on a Machine erroring out indefinitely. Is this designed to be a temporary delay to machine deletion?